crates/pontos/src/lib.rs

```diff
@@ -330,14 +354,21 @@ impl<S: Storage, C: StarknetClient, E: EventHandler + Send + Sync> Pontos<S, C,
                 )
                 .await?;
 
-            let blocks_events = self
+            let blocks_events = match self
                 .client
                 .fetch_events(
                     BlockId::Number(current_u64),
                     BlockId::Number(current_u64),
                     self.event_manager.keys_selector(),
                 )
-                .await?;
+                .await
+            {
+                Ok(events) => events,
+                Err(e) => {
+                    error!("Error while fetching events: {:?}", e);
```

Comment:
Here or we need a sleep, or we advance the block number. I would be on the side to have a sleep before continuing, or add several attempts as we did on the timestamp. WDYT?